source/common/ssl/context_impl.cc

```diff
@@ -223,6 +223,22 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
   return verified;
 }
 
+bool ContextImpl::uriMatch(const std::string& uriPattern, const char *uri) {
+  if (uriPattern == uri) {
+    return true;
+  }
+
+  size_t pattern_len = uriPattern.length();
+  if (pattern_len > 1 && uriPattern[pattern_len - 1] == '*' && uriPattern[pattern_len - 2] == '/') {
+    size_t uri_len = strlen(uri);
```

Comment:
agree that the code is not very efficient. Will rewrite so that it only takes one walk through both strings.